test/recaptcha.spec.js

```diff
@@ -88,6 +88,70 @@ describe("ReCAPTCHA", () => {
     instance._internalRef.current.execute();
     expect(grecaptchaMock.execute).toBeCalledWith(WIDGET_ID);
   });
+  it("executeAsync, should call grecaptcha.execute with the widget id", () => {
+    const WIDGET_ID = "someWidgetId";
+    const grecaptchaMock = {
+      render() {
+        return WIDGET_ID;
+      },
+      execute: jest.fn(),
+    };
+    // wrapping component example that applies a ref to ReCAPTCHA
+    class WrappingComponent extends React.Component {
+      constructor(props) {
+        super(props);
+        this._internalRef = React.createRef();
+      }
+      render() {
+        return (
+          <div>
+            <ReCAPTCHA
+              sitekey="xxx"
+              size="invisible"
+              grecaptcha={grecaptchaMock}
+              onChange={jest.fn()}
+              ref={this._internalRef}
+            />
+          </div>
+        );
+      }
+    }
+    const instance = ReactTestUtils.renderIntoDocument(React.createElement(WrappingComponent));
+    instance._internalRef.current.execute();
+    expect(grecaptchaMock.execute).toBeCalledWith(WIDGET_ID);
+  });
+  it("executeAsync, should return a promise", () => {
+    const WIDGET_ID = "someWidgetId";
+    const grecaptchaMock = {
+      render() {
+        return WIDGET_ID;
+      },
+      execute: jest.fn(),
+    };
+    // wrapping component example that applies a ref to ReCAPTCHA
+    class WrappingComponent extends React.Component {
+      constructor(props) {
+        super(props);
+        this._internalRef = React.createRef();
+      }
+      render() {
+        return (
+          <div>
+            <ReCAPTCHA
+              sitekey="xxx"
+              size="invisible"
+              grecaptcha={grecaptchaMock}
+              onChange={jest.fn()}
+              ref={this._internalRef}
+            />
+          </div>
+        );
+      }
+    }
+    const instance = ReactTestUtils.renderIntoDocument(React.createElement(WrappingComponent));
+    let result = instance._internalRef.current.executeAsync();
+    expect(result).toBeInstanceOf(Promise);
```

Comment:
💭 add an addition `expect` to test that the resolved value should be the `token`?

Reply:
Hmm, sorry if I am missing something, how would I get the test value of `token` to compare against?

Reply:
Hey @stemsmit , looks like there should be a couple different ways via the jest docs.
- https://jestjs.io/docs/en/asynchronous

`async/await` should work too, but i think i see what you're talking about. We'll need to setup a slightly sophisticated `grecaptchaMock`...

```
const WIDGET_ID = "someWidgetId";
const TOKEN = "someToken";
const grecaptchaMock = {
      render(_, { callback }) {
        this.callback = callback;
        return WIDGET_ID;
      },
      execute() { this.callback(TOKEN) },
    };
...
let result = instance._internalRef.current.executeAsync();
const retrievedToken = await result;
expect(retrievedToken).toBe(TOKEN);
```

Reply:
I attempted to use `async / await` but ran into an issue with `regeneratorRuntime` rather than make changes to fix that issue I set it up to return the promise the with the expect inside it.